### DISCUSSION TITLE
Grid: Responsive columns (`span`) performance

### DISCUSSION THREAD

**Discussion by matronator** [mantinedev/mantine#5619] [Q&A] [answered]
I read the docs page about styles performance and that responsive styles (the ones that you write like this: `{base: X, md: X, lg: X}`) are not very performant. I was wondering if the same applies to the `span` prop on the `<Grid.Col>` component. Is there a performance penalty if I write it like this:

```jsx
<Grid.Col span={{ base: 12, md: 6 }}></Grid.Col>
```

or does it only apply to styles?

Thanks!

**Comment by rtivital** [comment, marked as answer]
The logic is the same for all responsive styles and props. Do not render a thousand of items with responsive props and it will be fine.
